workflow/artifacts/http/http.go

```diff
@@ -17,6 +18,11 @@ type HTTPArtifactDriver struct{}
 func (h *HTTPArtifactDriver) Load(inputArtifact *wfv1.Artifact, path string) error {
 	// Download the file to a local file path
 	args := []string{"-fsS", "-L", "-o", path, inputArtifact.HTTP.URL}
+	headers := inputArtifact.HTTP.Headers
+	for _, v := range headers {
+		// Build curl -H string for each key-value header parameter
+		args = append(args, fmt.Sprintf("-H '%s: %s'", v.Name, v.Value))
```

Comment:
does this actually work? i feel like it should be

```golang
args = append(args, "-H", fmt.Sprintf('%s: %s'", v.Name, v.Value))
```

Reply:
You're right - the test I added didn't catch that since it just adds one header 😞

Reply:
Sorry, actually, you are *wrong*! Ha. I updated the test nevertheless to add 2 headers to the request.

```
"time="2020-09-15T16:17:13-04:00" level=info msg="curl -fsS -L -o /tmp/found-with-request-headers https://github.com/argoproj/argo -H 'Accept: application/json' -H 'Authorization: Bearer foo-bar'"
                                time="2020-09-15T16:17:13-04:00" level=info
```

Reply:
I'm not convinced (though can be obvs) can I please ask you you make sure you check this by starting up the controller and running a workflow that uses it?

Reply:
Happily - is the "starting up the controller" part documented somewhere?

We use/run Argo workflow controller at my job so I know how to submit workflows but haven't ever started up the process... please excuse the potentially dumb question/my ignorance on this!

Reply:
ah...

If you run `make start` it will install a test set-up to the `argo` namespace of your cluster and the UI will run on port 2746. 

Make sure you are not connected to your production system when you do it! 

I would recommend using something like Docker for Desktop for that.

You can then submit workflows.


You can see how to do this at 21:55 of the [Argo Workflow Contributor Workshop video](https://argo.ip-dl-mlp-ppd-usw2.iks.a.intuit.com/)

Reply:
@alexec link seems to be broken :(